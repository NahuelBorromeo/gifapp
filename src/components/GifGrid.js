import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    //El useEffect nos va a permitir ejecutar cierto codigo de manera condicional


    //usamos nuestro custom hook
    const { data:images, loading } = useFetchGifs( category );

    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ])
    


    return (
        <>
            <h3 className='animate__animated animate__fadeIn' >{ category }</h3>

            { loading && <p className='animate__animated animate__flash'>Loading</p> }

            <div className='card-grid'>
                    { 
                        images.map( img => 
                            <GifGridItem 
                                    key={ img.id }
                                    { ...img }
                            />)
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
