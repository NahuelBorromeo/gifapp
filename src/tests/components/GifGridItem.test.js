import { shallow } from "enzyme";
import React from "react";
import { GifGridItem } from "../../components/GifGridItem"

describe('testing en el componente GifGridItem', () => {
    
    const title = 'Gif de prueba';
    const url = 'https://localhost/algo.jpg';    
    const wrapper = shallow(
        <GifGridItem 
            title={title}
            url={url}
        />
        );

    test('should to show <GifGridItem /> correctly', () => {
                
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should to have a <p> with the title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });
    
    test('should to have a img src and alt equal from props', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        
    });

    test('should to have the classname animate__zoomIn', () => {
        
        const div = wrapper.find('div');

        const className = div.prop('className');

        expect( className.includes('animate__zoomIn') ).toBe(true);
        
    })
    


})
