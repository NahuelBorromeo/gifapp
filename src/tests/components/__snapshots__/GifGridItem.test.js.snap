// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en el componente GifGridItem should to show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Gif de prueba"
    src="https://localhost/algo.jpg"
  />
  <p>
    Gif de prueba
  </p>
</div>
`;
